test/v1/services/CreateForeignPropertyServiceSpec.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.services
+
+import support.UnitSpec
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.http.HeaderCarrier
+import v1.controllers.EndpointLogContext
+import v1.mocks.connectors.MockCreateForeignPropertyConnector
+import v1.models.errors._
+import v1.models.outcomes.ResponseWrapper
+import v1.models.request.createForeignProperty.{CreateForeignPropertyRequestBody, CreateForeignPropertyRequestData, ForeignFhlEea, ForeignFhlEeaExpenditure, ForeignFhlEeaIncome, ForeignProperty, ForeignPropertyExpenditure, ForeignPropertyIncome, RentIncome}
+import v1.models.response.create.CreateForeignPropertyResponse
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class CreateForeignPropertyServiceSpec extends UnitSpec {
+
+  val taxYear = "2018-04-06"
+  val nino = Nino("AA123456A")
+  private val correlationId = "X-123"
+
+  val body = CreateForeignPropertyRequestBody(
+    "2020-01-01",
+    "2020-01-31",
+    Some(ForeignFhlEea(
+      ForeignFhlEeaIncome(5000.99, Some(5000.99)),
+      Some(ForeignFhlEeaExpenditure(
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99),
+        Some(5000.99)
```

Comment:
Another mention of the body not technically being right due to it being a body which would never make it through